### PR TITLE
Add size prop and large variant to Testimonial

### PR DESCRIPTION
**Pull request by danielguillan**
## Summary

Adds a new `size` prop to the Testimonial component to allow users to set the quote's text size to either `small` (default) or `large`.

## List of notable changes:

- Added a `size` prop 
- Updated storybook and docs.

## What should reviewers focus on?

- [Storybook](https://primer-af7e2b4f24-26139705.drafts.github.io/storybook/?path=/story/components-testimonial--large)
- [Documentation](https://primer-af7e2b4f24-26139705.drafts.github.io/components/Testimonial)

## Supporting resources (related issues, external links, etc):

- https://github.com/github/primer/issues/2065
- [Component spec in Figma](https://www.figma.com/file/BJ95AjraesmRCWsKA013GS/branch/AoTaMS8g9VziqMVGOtzUOa/Primer-Brand?node-id=1852-27521&mode=design&t=NfOT4mXSWrGX4Hnb-11)
- [Design reference from 2023 Refactor](https://www.figma.com/file/gcLvL8UiWo3aI1qKQ422oO/2023-Refactor?node-id=1103-164248&mode=design&t=WUAh6f5gZzvSwaDa-11). _Note that this PR intentionally only adds the larger quote font-size but retains all other existing styling and layout._

## Contributor checklist:

- [x] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [x] Any drop in coverage, breaking changes or regressions have been documented above
- [x] New visual snapshots have been generated / updated for any UI changes
- [x] All developer debugging and non-functional logging has been removed
- [x] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:

The new large size Testimonial quote

### CODE REVIEW

File: packages/react/src/Testimonial/Testimonial.module.css [hunk 1]
```diff
@@ -17,12 +17,24 @@
...
+  font-size: var(--brand-text-size-300);
   margin-top: calc(var(--base-size-16) * -1); /* negate grid gap */
   display: block;
 }
 
+.Testimonial--size-large .Testimonial-quote {
+  font-size: var(--brand-text-size-700);
+  font-weight: var(--base-text-weight-semibold);
+  letter-spacing: -0.03em;
+  line-height: 1.2; /* requires custom line-height */
```
Comment (josepmartins) [inline packages/react/src/Testimonial/Testimonial.module.css:35]: should we use `rem` values here? 

Looks like the jump between `--brand-text-lineHeight-600` (2.75rem) to `--brand-text-lineHeight-700` (4rem) is a bit inconsistent, not sure if we should adapt  the `--brand-text-lineHeight-700` value at some point.

File: apps/docs/content/components/Testimonial.mdx [hunk 2]
```diff
@@ -57,7 +57,7 @@ A logo (or similar) visual can be provided as an alternative to avatars.
```
Comment (josepmartins) [inline apps/docs/content/components/Testimonial.mdx:57]: non-blocking: The link to the storybook seems to be broken, can we change it to?

```html
/brand/storybook/?path=/story/components-Testimonial--playground
```

**Review by josepmartins — APPROVED** [review]
LGTM! 

left a couple of non-blocking comments

File: packages/react/src/Testimonial/Testimonial.module.css (hunk 1 above)
Comment (danielguillan) [reply packages/react/src/Testimonial/Testimonial.module.css:35]: Yeah, we'll need to review the line-height scale include a closer value.

File: .changeset/three-kids-pay.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@primer/react-brand': minor
```
Comment (rezrah) [inline .changeset/three-kids-pay.md:2]: This can be a patch.

```suggestion
'@primer/react-brand': patch
```

File: .changeset/three-kids-pay.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+---
+
+Adds a `size` prop to control Testimonial's quote text size
```
Comment (rezrah) [inline .changeset/three-kids-pay.md:5]: Nit: 

```suggestion
Added a `size` prop to control Testimonial's text size
```

File: packages/react/src/Testimonial/Testimonial.module.css [hunk 5]
```diff
@@ -17,12 +17,24 @@
...
   font-family: var(--brand-fontStack-sansSerif);
+  font-size: var(--brand-text-size-300);
   margin-top: calc(var(--base-size-16) * -1); /* negate grid gap */
   display: block;
 }
 
+.Testimonial--size-large .Testimonial-quote {
+  font-size: var(--brand-text-size-700);
+  font-weight: var(--base-text-weight-semibold);
+  letter-spacing: -0.03em;
```
Comment (rezrah) [inline packages/react/src/Testimonial/Testimonial.module.css:34]: FYI, we have a token for this..

```suggestion
  letter-spacing: var(--brand-text-letterSpacing-900)
```

**Comment by danielguillan** [comment]
> Looks great ✨
> 
> Couldn't see a corrosponding test, so would be great to add one if possible? We may risk seeing a drop in coverage otherwise.

@rezrah Testimonial has no tests at all, that's why I couldn't add one for `size`. Happy to work on a follow-up PR that adds coverage for this component.

**Pull request merged**